wagtail/admin/views/pages/bulk_actions/publish.py

```diff
@@ -32,19 +32,15 @@ def get_execution_context(self):
         }
 
     @classmethod
-    def execute_action(cls, objects, **kwargs):
-        include_descendants = kwargs.get('include_descendants', False)
-        user = kwargs.get('user', None)
-        if user is None:
-            return
+    def execute_action(cls, objects, include_descendants=False, user=None):
         for page in objects:
             revision = page.save_revision(user=user)
             revision.publish(user=user)
             cls.num_parent_objects += 1
 
             if include_descendants:
                 for draft_descendant_page in page.get_descendants().not_live().defer_streamfields().specific():
-                    if draft_descendant_page.permissions_for_user(user).can_publish():
+                    if user is None or draft_descendant_page.permissions_for_user(user).can_publish():
```

Comment:
At first this bit here feelt a tad 🤔 but it works with the idea that `{type}BulkAction.execute_action()` will just do the thing if called directly in code. The same applies to Page `publish()`, so all good!